### PR TITLE
removes tesla

### PR DESCRIPTION
**Pull request by AnCopper**
#5057

## Changelog
:cl:
remove: tesla generator and related machinery have been removed.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
This is _stupid_. I thought we aren't /tg/? if people get's butthurt for tesla's getting out of control so frequently. that's an administrative problem.

If the defense it's that it's OP and produces a lot of power. TEG makes wayyy more power while also idiot proof while also be able to be constructed early-midround if you yell at the scientist hard enough. Tesla's setups (safe ones that is) takes a long while to set up. most likely in the 1:30 mark and produces wayy fewer power than the TEG.

Don't remove things because the admin team can't handle the griefing by it. that's their problem. no reason removing fun and cool stuff just because people are using it to grief. you guys are MRP you should be able to handle some griefers here and there before they destroy the entire station.

Oh. it bugs out the server... just disable the ability to purchase it in the cargo console instead of removing the whole thing all-together. they still might be used for special events or prayers.

**Review by PowerfulBacon — COMMENTED** [review]
I think this would be 100x better if you just limitted the teslas maximum ball size to prevent it from making a laggy amount of shocks.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Provide some kind of reason for why its good. Randomly removing something with no reason to why it should be removed won't get merged (Unless its something critically breaking the game, and if so then that's the reason for its removal and should be stated).

**Comment by AnCopper** [comment]
> Provide some kind of reason for why its good. Randomly removing something with no reason to why it should be removed won't get merged (Unless its something critically breaking the game, and if so then that's the reason for its removal and should be stated).

The reason is that your head developer wants it.

**Comment by utilbird** [comment]
@Singularity69 It's for performance reasons, not any of the points you listed. Making it more rare doesn't mean it's not going to grind the server to a halt.

With that being said, there are much better temporary solutions than just removing it. As bacon said, limiting the ball count is an effective way to deal with it. The best possible option would be to rework how balls work so it's not spamming object instances with all of their own process, but that's effort. 

👎

**Comment by MNarath1** [comment]
imao this also touches the sm tesla reaction removing a lot of important stuff without actually filling the gap not even adressing said issue at all

**Comment by MNarath1** [comment]
like why is it even removing sm tesla at all unlike normal tesla sm tesla usually won't last long and is pretty much a guranteed round ender of course it also lags a lot but ... so does the singularity if it breaks loose in both cases it will end the round pretty quick on the other side this doesn't seem to remove the tesla bounty at all hmmm.

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -302,15 +302,6 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			var/obj/singularity/S = new(T)
 			S.energy = 800
 			S.consume(src)
-	else
-		investigate_log("has exploded.", INVESTIGATE_ENGINES)
-		explosion(get_turf(T), explosion_power * max(gasmix_power_ratio, 0.205) * 0.5 , explosion_power * max(gasmix_power_ratio, 0.205) + 2, explosion_power * max(gasmix_power_ratio, 0.205) + 4 , explosion_power * max(gasmix_power_ratio, 0.205) + 6, 1, 1)
```
Comment (PowerfulBacon) [inline code/modules/power/supermatter/supermatter.dm:307]: imagine removing SM explosion

Comment (AnCopper) [reply code/modules/power/supermatter/supermatter.dm:307]: Oh.

**Comment by AnCopper** [comment]
> This is _stupid_. I thought we aren't /tg/? if people get's butthurt for tesla's getting out of control so frequently. that's an administrative problem.
> 
> If the defense it's that it's OP and produces a lot of power. TEG makes wayyy more power while also idiot proof while also be able to be constructed early-midround if you yell at the scientist hard enough. Tesla's setups (safe ones that is) takes a long while to set up. most likely in the 1:30 mark and produces wayy fewer power than the TEG.
> 
> Don't remove things because the admin team can't handle the griefing by it. that's their problem. no reason removing fun and cool stuff just because people are using it to grief. you guys are MRP you should be able to handle some griefers here and there before they destroy the entire station.
> 
> Oh. it bugs out the server... just disable the ability to purchase it in the cargo console instead of removing the whole thing all-together. they still might be used for special events or prayers.

I have no idea what you are rambling about, this pull request removes the tesla generator and the grounding rods/tesla coils.

**Comment by Archanial** [comment]
Why not port tg's tesla refactor thingy instead?

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -305,12 +305,6 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	else
 		investigate_log("has exploded.", INVESTIGATE_ENGINES)
 		explosion(get_turf(T), explosion_power * max(gasmix_power_ratio, 0.205) * 0.5 , explosion_power * max(gasmix_power_ratio, 0.205) + 2, explosion_power * max(gasmix_power_ratio, 0.205) + 4 , explosion_power * max(gasmix_power_ratio, 0.205) + 6, 1, 1)
-		if(power > POWER_PENALTY_THRESHOLD)
-			investigate_log("has spawned additional energy balls.", INVESTIGATE_ENGINES)
-			var/obj/singularity/energy_ball/E = new(T)
-			E.energy = power
-		qdel(src)
-
 //this is here to eat arguments
```
Comment (PowerfulBacon) [inline code/modules/power/supermatter/supermatter.dm]: .

Comment (AnCopper) [reply code/modules/power/supermatter/supermatter.dm:308]: Hmm?

Comment (PowerfulBacon) [reply code/modules/power/supermatter/supermatter.dm]: Test it lol

Comment (AnCopper) [reply code/modules/power/supermatter/supermatter.dm:308]: I'm not sure what exactly you want me to test here.

Comment (MNarath1) [reply code/modules/power/supermatter/supermatter.dm]: line 312 look at what you are removing

Comment (AnCopper) [reply code/modules/power/supermatter/supermatter.dm:308]: It's to delete the Supermatter Crystal during the tesla delam.

**Comment by ike709** [comment]
> Why not port tg's tesla refactor thingy instead?

WYCI

**Comment by ike709** [comment]
But yes, either make the tesla not grind the server to a halt or remove it. I don't particularly care which.

**Comment by MNarath1** [comment]
yeah currently looking on working on a port of that trough there seem to be some differences that i have to investigate beforehand so that i don't accidentally break something

**Comment by AnCopper** [comment]
Should be good now. @ike709

**Comment by PowerfulBacon** [comment]
This pr is extremely half assed and still keeps stuff like the co2 sm interaction and messages without it doing anything. The fix PR is 2 lines long. I dont want to be rude but I dont have faith that the code is tested or that any effort has been put into this PR, it has grudge code levels of effort. I'll port the 2 line fix later and close this PR.

**Comment by AnCopper** [comment]
> This pr is extremely half assed and still keeps stuff like the co2 sm interaction and messages without it doing anything. The fix PR is 2 lines long. I dont want to be rude but I dont have faith that the code is tested or that any effort has been put into this PR, it has grudge code levels of effort. I'll port the 2 line fix later and close this PR.

Geez, sorry that I offended you that much.

**Comment by Archanial** [comment]
:key: improve not remove smh!!!!

**Comment by qwertyquerty** [comment]
🔑

**Pull request closed without merging**